sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/plans/logical/basicOperators.scala

```diff
@@ -191,6 +191,8 @@ case class Expand(
     val sizeInBytes = child.statistics.sizeInBytes * projections.length
     Statistics(sizeInBytes = sizeInBytes)
   }
+
+  override def missingInput = super.missingInput.filter(_.name != VirtualColumn.groupingIdName)
```

Comment:
@liancheng The grouping ID virtual column only used in Expand(LogicalPlan), so I think it's better to add the grouping ID virtual column in subclass(Expand) of QueryPlan.